spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Let's separate two things: How to use the lib in boot and how to use it as standalone

So, the standalone lib, or the template, should just have a dependency on credentials and project id.

How they get populated, it will depend on how the user is using SI. For example someone may still be relying on XML to configure the beans, and it may turn out that they use this:
```
<bean id="pubsubTemplate" class "...">
<property name="credentials" ref""/>
<property name="projectId" value="mygooglecloudprojectid"/>
</bean>```

This is an example, I know we are using constructors, it's just to show that how someone creates a credential object or pass the project is highly dependable on how they are configuring their spring project.

Boot users, on the other hand, can benefit from autoconfiguration (many times they choose not to BTW) and that's where GCPProperties live.

Does it makes sense now?